cpp/src/arrow/util/compression_lz4.cc

```diff
@@ -39,8 +39,14 @@ static Status LZ4Error(LZ4F_errorCode_t ret, const char* prefix_msg) {
   return Status::IOError(prefix_msg, LZ4F_getErrorName(ret));
 }
 
+static LZ4F_preferences_t DefaultPreferences() {
+  LZ4F_preferences_t prefs;
+  memset(&prefs, 0, sizeof(prefs));
```

Comment:
Nit: there's `LZ4F_INIT_PREFERENCES` (which also zeros every field), but we don't really need to care about this for now